### ISSUE TITLE
Napoleon: Handling of colons in summary lines in inline literals

### ISSUE THREAD

**Issue by ksunden** [sphinx-doc/sphinx#4016]
Subject: Napoleon: Handling of colons in summary lines in inline literals

### Problem
- Colons in inline literals (with two backticks) are transformed into ` -- ` when parsed with napoleon
   - When using single backticks, it still does not show up as a colon, rather ` - `

#### Procedure to reproduce the problem
Have the following method and docstring in your code (seemingly any colon within an inline literal on the summary line will do):
```
def hms(self):
        """Get time formated as ``HH:MM:SS``."""
        return "Return val is irrelevant here"
```
And build the documentation using 
```
$ make html
```

Note that if any of the following modifications are made, this particular docstring will turn out as expected (though not necessarily as desired, and these changes may break other docstrings in the project):
- Napoleon extension is removed in `conf.py`
- numpydoc used in place of napoleon
- two colons are placed in a row in the docstring (both are displayed as colons)
- The inline literal does not appear on the summary line, but rather in the body of the docstring

Note also that removing the inline literal entirely results in the following error (Error references the next docstring):
```
WARNING: Inline strong start-string without end-string.
```
and produces the following in the html:
```
**Get time formated as HH* – MM* – SS.
```
(With the `**` being a link pointing to a non existent `id1` tag on the same page)
This particular error may be a separate bug, but including here because I found it in my testing.

#### Error logs / results
```
No error is produced 
```
- http://wright.tools/en/development/api/WrightTools.kit.html#WrightTools.kit.TimeStamp.hms

#### Expected results
The colons in inline literals should show up as colons i.e. in this example:

Get time formated as `HH:MM:SS`.

### Reproducible project / your project
- https://github.com/wright-group/WrightTools/tree/hmsColon

- See github issue wright-group/WrightTools#183

### Environment info
- OS: Arch Linux
- Python version: 3.6.2
- Sphinx version: 1.6.3
- Problem exhibited in both Firefox and Chromium

EDIT: Updated the link to be to a branch specifically made to preserve this issue, as I implement a workaround (putting the literal including colons on the next line)

**Comment by goodmami** [comment]
Does it only happen in summary lines for you? I see it elsewhere, too. E.g.:

```python
def makedict(key, value):
    """Put *key* and *value* into a dictionary.

    Returns:
        A dictionary ``{key: value}``
    """
    return {key: value}
```

**Comment by goodmami** [comment]
As with #4021, closer inspection reveals that this issue is slightly different than mine, but in this case I think the reason is the same. The Napoloen code uses greedy regexes to look for colons (search for `_partition_field_on_colon` in [ext/napoleon/docstring.py](https://github.com/sphinx-doc/sphinx/blob/master/sphinx/ext/napoleon/docstring.py)). In my case, replacing the `Returns:` block with one of the following resolves the problem:

    Returns:
        A dictionary :code:`{key: value}`

or

    Returns:
        dict: ``{key: value}``

The `docstring.py` code may need to be updated to look for inline formatting and ignore those chunks when splitting on the colon.

**Comment by ksunden** [comment]
Went to revisit this, to see if changes which didn't reference this issue may have changed behaviour:

It seems like they have _changed_, though still produces unexpected results:

this now produces the following html

```html

<dt id="WrightTools.kit.TimeStamp.hms">
<code class="descname">hms</code><a class="headerlink" href="#WrightTools.kit.TimeStamp.hms" title="Permalink to this definition">¶</a></dt>
<dd><p>SS``.</p>
<table class="docutils field-list" frame="void" rules="none">
<colgroup><col class="field-name">
<col class="field-body">
</colgroup><tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">Get time formated as <a href="#id1"><span class="problematic" id="id2">``</span></a>HH</td>
</tr>
<tr class="field-even field"><th class="field-name">Type:</th><td class="field-body">MM</td>
</tr>
</tbody>
</table>
</dd>
```

With Sphinx v1.8.2

It also produces the following error output:

```
<filepath>:docstring of WrightTools.kit.TimeStamp.hms:3: WARNING: Inline literal start-string without end-string.
```

Previously, no warning was printed

**Comment by tk0miya** [comment]
It seems this has already been fixed in v3.4.2. This is a screenshot for the given code:

<img width="369" alt="スクリーンショット 2021-01-27 0 41 48" src="https://user-images.githubusercontent.com/748828/105867545-8f16c000-6038-11eb-9f67-09617332102a.png">

So I'm closing this. Please file a new issue if you're still in trouble.

Thanks,
